build: intermittent race condition when building with parallelization

### Terraform Version
0.8.7

### Problem Description
I couldn't find a previous issue on this, so I apologize if it's a duplicate.

When building in parallel (makeflag -j4), there is an intermittent race condition where the generated file `./command/internal_plugin_list.go` isn't gofmt'd before the build steps take place. It presents itself pretty consistently in Homebrew builds on El Capitan.

While I realize that Homebrew builds from source are not preferred by Hashicorp (as opposed to the signed binaries), and this issue is likely low priority, I wanted to report it in the event that it was correctable in some simple way.

I think the cause of this is the `bin` make target has `generate` and `fmtcheck` as dependencies, and when these run at the same time they can trip up on each other.

### Debug Output
```
==> make bin
go generate $(go list ./... | grep -v /terraform/vendor/)
==> Checking that code complies with gofmt requirements...
2017/02/17 22:20:54 Generated command/internal_plugin_list.go
gofmt needs running on the following files:
./command/internal_plugin_list.go
You can use the command: `make fmt` to reformat code.
make: *** [fmtcheck] Error 1

==> Formula
Path: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/terraform.rb
==> Configuration
HOMEBREW_VERSION: 1.1.10-21-ga74b7ade6
ORIGIN: https://github.com/Homebrew/brew
HEAD: a74b7ade66e340c55b3bea7b6226fa68a311b4ce
Last commit: 18 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core/
Core tap HEAD: 50ecc9418bc49b3d65aaf62f26dc69ec519ae6fa
Core tap last commit: 4 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit nehalem
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.11.1 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_112
macOS: 10.11.6-x86_64
Xcode: 8.2.1
CLT: 8.2.0.0.1.1480973914
X11: 2.7.9 => /opt/X11
==> ENV
HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
SDKROOT: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk
MAKEFLAGS: -j4
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /usr/include/libxml2:/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
MACOSX_DEPLOYMENT_TARGET: 10.11
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.11
PATH: /usr/local/Homebrew/Library/Homebrew/shims/super:/usr/local/opt/go/bin:/usr/local/opt/git/bin:/usr/bin:/bin:/usr/sbin:/sbin:/private/tmp/terraform-20170217-44441-1gw0bj/terraform-0.8.7
```

### Steps to Reproduce
1. `export MAKEFLAGS=-j4`
2. `export XC_OS=darwin`
3. `export XC_ARCH=amd64`
4. `make bin`

### References
- Homebrew/homebrew-core#10010

--- a/Makefile
+++ b/Makefile
@@ -104,4 +104,9 @@ errcheck:
 vendor-status:
 	@govendor status
 
+# disallow any parallelism (-j) for Make. This is necessary since some
+# commands during the build process create temporary files that collide
+# under parallel conditions.
+.NOTPARALLEL:
+
 .PHONY: bin core-dev core-test cover default dev errcheck fmt fmtcheck generate plugin-dev quickdev test-compile test testacc testrace tools vendor-status vet

